preprocessing.py

```diff
@@ -56,10 +69,26 @@ def deserializeIndex(filePath):
         all_terms_in_dict = pickle.loads(s)
         return all_terms_in_dict
 
+def sliding_window(size_of_window, docid, dictionary_of_all_terms):
+    with open(docid, 'r') as my_file:
+        data = my_file.read().split()
+    k = len(data) // size_of_window
+    dimension = len(dictionary_of_all_terms)
+    result = []
+    for l in range(0, k):
+        current_vector = sparse.coo_matrix((dimension, 1))
+        for e in range(0, size_of_window):
+            current_vector += sparse_basis_vector_creation(dictionary_of_all_terms[data[e]], dimension)
+        result.append(current_vector)
+    return result
+
 
 if __name__ == '__main__':
     path = 'cacm.ml/*.xml'
     # path = sys.argv[1]
     list_of_fields = ["TITLE"]
     preprocessing(path, list_of_fields)
-    print(deserializeIndex("all_terms_in_dict"))
+    all_terms = deserializeIndex("all_terms_in_dict")
+#    print(all_terms)
+    dest = "preprocessed_files"
+    sliding_window(1, dest + "/" + "1", all_terms)
```

Comment:
you can create indexes and sliding windows on the way during preprocessing step
maybe you try to integrate steps between them saving abstraction of methods? 
one of solution can be creating global vars with dictionaries and indexes and reusing of them between steps
there are too many excess file-open operations